docs/adr/0005-arc-pointers.md

```diff
@@ -0,0 +1,167 @@
+# Use raw `Arc<T>` pointers to pass objects across the FFI
+
+* Status: proposed
+* Deciders: mhammond, rfkelly
+* Consulted: travis, jhugman, dmose
```

Comment:
@travis79, @jhugman, @dmose do you feel like you have been sufficiently consulted on this change, or would you like additional details or time or otherwise to understand the implications here in more detail?